pages/dashboard.vue

```diff
@@ -11,34 +11,86 @@
 </template>
 
 <script lang="ts">
-import { defineComponent } from '@nuxtjs/composition-api'
-import { useResult, useQuery } from '@vue/apollo-composable'
+import { useQuery, useResult } from '@vue/apollo-composable'
+import { defineComponent, onMounted, onUnmounted } from '@vue/composition-api'
 import { gql } from '~/apollo'
 import { useUiStore } from '~/store'
 
+const FIRST = 4
+
 export default defineComponent({
   middleware: ['authenticated'],
 
   setup() {
     const ui = useUiStore()
 
-    const { result } = useQuery(
+    const { result, fetchMore } = useQuery(
       gql(/* GraphQL */ `
-        query GetDocuments($groupId: ID, $query: String) {
+        query GetDocuments(
+          $groupId: ID
+          $query: String
+          $first: Int
+          $after: String
+        ) {
           me {
             id
-            documents(filterBy: { groupId: $groupId, query: $query }) {
-              ...DocumentForView
+            documents(
+              filterBy: { groupId: $groupId, query: $query }
+              first: $first
+              after: $after
+            ) {
+              edges {
+                node {
+                  ...DocumentForView
+                }
+              }
+              pageInfo {
+                endCursor
+                hasNextPage
+              }
             }
           }
         }
       `),
       () => ({
         groupId: ui.selectedGroupId,
         query: ui.activeSearchQuery,
+        first: FIRST,
+        after: '',
       })
     )
-    const documents = useResult(result, null, (data) => data?.me?.documents)
+
+    const documents = useResult(result, null, (data) =>
+      data?.me?.documents.edges.map((edge) => edge.node)
+    )
+
+    const loadMoreDocuments = () => {
+      void fetchMore({
+        variables: {
+          groupId: ui.selectedGroupId,
+          query: ui.activeSearchQuery,
+          first: FIRST,
+          after: result.value?.me?.documents.pageInfo.endCursor,
+        },
+      })
+    }
+
+    const handleScroll = () => {
```

Comment:
I think its best to use an existing library to handle the infinite-loading (as there are quite a few edge cases to consider). https://github.com/tangbc/vue-virtual-scroll-list seems to be popular, see https://github.com/tangbc/vue-virtual-scroll-list/blob/master/example/src/views/infinite-loading/Main.vue for an example on how to use it.